### DISCUSSION TITLE
React 17: pragma and pragmaFrag cannot be set when runtime is automatic

### DISCUSSION THREAD

**Discussion by michrome** [mdx-js/mdx#1345] [Q&A] [answered]
I have been [enjoying MDX@2](https://github.com/mdx-js/mdx/issues/1170#issuecomment-729775035) but hit the error `pragma and pragmaFrag cannot be set when runtime is automatic` when trying to use it with React v17, i.e.

```
  "dependencies": {
    "@mdx-js/loader": "^2.0.0-next.8",
    "@next/mdx": "^10.0.2",
    "next": "^10.0.2",
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
  }
```

If I downgrade react to v16, then everything works fine 👌

```
  "dependencies": {
    "@mdx-js/loader": "^2.0.0-next.8",
    "@next/mdx": "^10.0.2",
    "next": "^10.0.2",
    "react": "^16.14.0",
    "react-dom": "^16.14.0"
  }
```

I am wondering If the issue relates to the [new JSX Transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) but I'm not sure to resolve. Thanks to anyone who can offer a suggestion!

**Comment by ChristianMurphy** [comment, marked as answer]
V1 got a fix for this https://github.com/mdx-js/mdx/pull/1307
it looks like it hasn't been applied to v2 yet https://github.com/mdx-js/mdx/blob/e194fc1a61715549be10d56055ea4ad25533fcb1/packages/mdx/index.js#L95-L109
A PR porting the change to v2 would be welcome!

**Reply by michrome** [reply]
Thanks for the quick reply @ChristianMurphy . Looks like you've already approved my PR #1346 🎉
